examples/error/main.go

```diff
@@ -0,0 +1,508 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"time"
+
+	dbsql "github.com/databricks/databricks-sql-go"
+	dbsqlctx "github.com/databricks/databricks-sql-go/driverctx"
+	dbsqlerr "github.com/databricks/databricks-sql-go/errors"
+	"github.com/joho/godotenv"
+)
+
+// NOTE: these tests print errors to the console and are for demonstration purposes, not unit test coverage
```

Comment:
we need to make this file into a real example of how to use the error interfaces we are exposing. I am assuming that this file was not meant to be merged as is.